pkg/capr/planner/probe.go

```diff
@@ -4,73 +4,80 @@ import (
 	"fmt"
```

Comment:
Scalability might be a concern in this section; let's discuss potential optimizations.